compiler/vm/vm_enums.nim

```diff
@@ -70,6 +73,11 @@ type
     opcAddSeqElem,
     opcRangeChck,
 
+    # TODO: would it be better to NimNode specific opcodes and replace them with vm callback?
```

Comment:
I missed a "remove" in the TODO there, but I think you understood it anyway.

Positive effects would be:
- Minor postive, but adding new `NimNode` operations (.i.e. new procs to `std/macros`) gets simpler. Instead of having to adjust `vmgen`, adding a new opcode and then implement it in `vm.nim`, you'd just add a new callback in a single location (currently `vmhooks.nim`). 
- Less code in `vm.nim`. `NimNode` instructions account for a significant amount of code in `vm.nim`. Regardless of turning them into callbacks or not, their implementation could be moved into functions in a new module.
- Less code in execution loop -> better instruction cache usage
- Guest code not using `NimNode` instructions (non-macro code such as default NimScript) could run faster

Reply:
👍 Cool, looks like you've got all the inputs to make a sound decision.